src/System.Diagnostics.PerformanceCounter/src/System/Diagnostics/ProcessWaitHandle.Windows.cs

```diff
@@ -0,0 +1,34 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Microsoft.Win32.SafeHandles;
+using System.Runtime.InteropServices;
+using System.Threading;
+
+namespace System.Diagnostics
```

Comment:
It's an internal class. Should I make it not internal? Is there a way to make certain internal classes visible to some external libraries?

Reply:
You could move the file Dan mentioned into the Common folder and add a link to it from both projects (System.Diagnostics.Process and yours).

Reply:
Could I do the same thing with TraceInternal.cs so I can use the lock from both places?

Reply:
if your questions asks if you can share an objects state with linking files then the answer is no as you are compiling the linked code into both projects. That internal types then exists twice and can't share any information.

Reply:
Fixed my previous message. I hope now it's readable ;)

Reply:
> Is there a way to make certain internal classes visible to some external libraries?

Yes there is a way (see InternalsVisibleTo) but we don't want to use that in any of the published assemblies you are compiling against.

Reply:
Ok, that's what I was trying to figure out. Thanks. I think it is going to be a lot of work to move ProcessWaitHandle to Common, as there are different implementations for Unix and Windows, and it's used in files used on both platforms in Process (Process.cs, notably). In that instance, I can't just change it to Interop.Kernel32.ProcessWaitHandle. I could move the function that creates a ProcessWaitHandle to Process.Windows.cs and Process.Unix.cs (and then likely all the other process.<platform>.cs files), but that seems like it would ultimately be even more duplicate code that we'd like to avoid. Thoughts?